DNN Platform/Library/Services/OutputCache/Providers/MemoryProvider.cs

```diff
@@ -175,6 +187,99 @@ public override bool StreamOutput(int tabId, string cacheKey, HttpContext contex
         	return true;
         }
 
+        private void OnSynchronizeResponseCallback(IAsyncResult asynchronousResult)
+        {
+            // State of request is asynchronous.
+            var request = (HttpWebRequest)asynchronousResult.AsyncState;
+            try
+            {
+                var response = (HttpWebResponse)(request.EndGetResponse(asynchronousResult));
+
+                if (response == null || response.StatusCode != HttpStatusCode.OK)
+                {
+                    //Log Error
+                    Exceptions.Exceptions.LogException(new Exception("Synchronization Error: Url - " + request.RequestUri.AbsoluteUri + "; Status - " + response.StatusCode + " " + response.StatusDescription));
+                }
+            }
+            catch (WebException e)
+            {
+                if (e.Status != WebExceptionStatus.RequestCanceled)
+                {
+                    Exceptions.Exceptions.LogException(new Exception("Synchronization Error in Request: " + request.RequestUri.AbsoluteUri, e));
+                }
+            }
+            finally
+            {
+                _registeredWaitHandle?.Unregister(asynchronousResult.AsyncWaitHandle);
+            }
+        }
+
+        private static void TimeoutCallback(object state, bool timedOut)
+        {
+            if (timedOut)
+            {
+                var request = (HttpWebRequest)state;
+                request?.Abort();
+            }
+        }
+
+        private void SendMessage(int tabId)
+        {
+            //do not synchronize cache during upgrade process.
+            if (HttpContext.Current != null && IsUpgradeRequest(HttpContext.Current.Request))
+            {
+                return;
+            }
+
+            var servers = ServerController.GetServers();
+            if (servers.Count > 1)
+            {
+                var thisServer = servers.Single(s => s.ServerName == Globals.ServerName && s.IISAppName == Globals.IISAppName);
+
+                //Get the servers
+                var enabledServers = ServerController.GetEnabledServers().Where(s => !(s.ServerName == thisServer.ServerName
+                                                                            && s.IISAppName == thisServer.IISAppName)
+                                                                            && (s.ServerGroup == thisServer.ServerGroup || string.IsNullOrEmpty(thisServer.ServerGroup))
+                                                                            && !string.IsNullOrEmpty(s.Url));
+                //Iterate through the servers
+                foreach (var server in enabledServers)
+                {
+                    var dataParam = Host.DebugMode ? tabId.ToString() : UrlUtils.EncryptParameter(tabId.ToString(), Host.GUID);
+                    //build url
+                    var url = "http://" + synchronizeUrl;
+                    if (HostController.Instance.GetBoolean("UseSSLForCacheSync", false))
```

Comment:
I cannot find this key, is this a completely new setting?
Looks like something that should be derived from an existing setting instead?

Reply:
This is a setting that currently only exists at the Evoq level, as it is part of the WebRequestCachingProvider